gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/artifacts/DependencyDiscovery.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.artifacts;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.ArtifactView;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.transform.TransformAction;
+import org.gradle.api.artifacts.transform.TransformParameters;
+import org.gradle.api.artifacts.type.ArtifactTypeDefinition;
+import org.gradle.api.attributes.Attribute;
+import org.gradle.api.attributes.Category;
+import org.gradle.util.GUtil;
+
+public final class DependencyDiscovery {
+    public static final Attribute<String> ARTIFACT_FORMAT = Attribute.of("artifactType", String.class);
```

Comment:
Just for my own understanding, is this `artifactType` string a built in bit of gradle or something custom we've made?

I see it [in the docs](https://docs.gradle.org/current/userguide/variant_model.html#sec:variant-visual) (that I have just spent a good long time grokking!) for something attached to artifacts, but not sure if that's the same thing as a variant attribute?

Reply:
I see, it is a built in thing (just no public api to get the magic string/Attribute object) and just filters the things you want to resolve via the configuration.

Reply:
it is something that gradle defines, the variable is exposed as part of their internal API